x-pack/plugins/ml/public/application/jobs/new_job/common/job_creator/util/filter_runtime_mappings.test.ts

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Job, Datafeed } from '../../../../../../../common/types/anomaly_detection_jobs';
+import { filterRuntimeMappings } from './filter_runtime_mappings';
+
+function getJob(): Job {
+  return {
+    job_id: 'test',
+    description: '',
+    groups: [],
+    analysis_config: {
+      bucket_span: '15m',
+      detectors: [
+        {
+          function: 'mean',
+          field_name: 'responsetime',
+        },
+      ],
+      influencers: [],
+    },
+    data_description: {
+      time_field: '@timestamp',
+    },
+    analysis_limits: {
+      model_memory_limit: '11MB',
+    },
+    model_plot_config: {
+      enabled: false,
+      annotations_enabled: false,
+    },
+  };
+}
+
+function getDatafeed(): Datafeed {
+  return {
+    datafeed_id: 'datafeed-test',
+    job_id: 'dds',
+    indices: ['farequote-*'],
+    query: {
+      bool: {
+        must: [
+          {
+            match_all: {},
+          },
+        ],
+      },
+    },
+    runtime_mappings: {
+      responsetime_big: {
+        type: 'double',
+        script: {
+          source: "emit(doc['responsetime'].value * 100.0)",
+        },
+      },
+      airline_lower: {
+        type: 'keyword',
+        script: {
+          source: "emit(doc['airline'].value.toLowerCase())",
+        },
+      },
+    },
+  };
+}
+
+function getAggs() {
+  return {
+    buckets: {
+      date_histogram: {
+        field: '@timestamp',
+        fixed_interval: '90000ms',
+      },
+      aggregations: {
+        responsetime: {
+          avg: {
+            field: 'responsetime_big',
+          },
+        },
+        '@timestamp': {
+          max: {
+            field: '@timestamp',
+          },
+        },
+      },
+    },
+  };
+}
+
+describe('filter_runtime_mappings', () => {
+  describe('filterRuntimeMappings()', () => {
+    let job: Job;
+    let datafeed: Datafeed;
+    beforeEach(() => {
+      job = getJob();
+      datafeed = getDatafeed();
+    });
+
+    test('returns no runtime mappings, no mappings in aggs', async () => {
+      const resp = filterRuntimeMappings(job, datafeed);
+      expect(Object.keys(resp?.runtime_mappings ?? {}).length).toEqual(0);
+
+      expect(Object.keys(resp?.discarded_mappings ?? {}).length).toEqual(2);
```

Comment:
not sure if assertion in tests should contain nullish coalescing of any conditional logic. If you expect `resp` to be empty, it's sufficient to verify it's empty. If `runtime_mappings ` can be `undefined`, we need to explicitly check for undefined instead of falling back to an empty array and verify keys length. It's related to the entire file. Also, it seems like all test suits are synchronous, so `async` is redundant and might be misleading during reading.

Reply:
good spot, this was for an older version of the function. the response can't be undefined now.
updated the tests in 917effa0f28beceefbad252ef6fa58f5f1fcd529